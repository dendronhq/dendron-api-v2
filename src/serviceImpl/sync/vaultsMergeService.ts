
import fs from "fs";
import matter from "gray-matter";
import { Result, errAsync, ok } from "neverthrow";
import path from "path";
import { MergeVaultsRequest, MergeVaultsResponse } from "../../api/generated/api";
import { logger } from "../../logger";

interface Counts {
  allFiles: number;
  merged: number;
}

export class VaultsMergeService {
  async execute(opts: MergeVaultsRequest): Promise<Result<MergeVaultsResponse, Error>> {
    const ctx = "VaultsMergeService"
    const { src, dest } = opts;
    if (!fs.existsSync(src)) {
      logger.error(`Source directory ${src} does not exist.`);
      return errAsync(new Error(`Source directory ${src} does not exist.`))
    }
    if (!fs.existsSync(dest)) {
      logger.error(`Destination directory ${dest} does not exist.`);
      return errAsync(new Error(`Destination directory ${dest} does not exist.`))
    }
    logger.info({ ctx, msg: "enter", opts });

    const extraMetadataOnTarget = opts.extraMetadata ?? {
    };
    const resp = await this.mergeFiles({ srcDir: src, destDir: dest, dryRun: false, extraMetadataOnTarget })
    return ok({
      numNotesMerged: resp.counts.merged,
      numNotesNew: resp.counts.allFiles - resp.counts.merged,
    });
  }

  mergeFiles({
    extraMetadataOnTarget,
    srcDir,
    destDir,
    dryRun = false
  }: {
    extraMetadataOnTarget: any,
    srcDir: string,
    destDir: string,
    dryRun: boolean
  }): { merged: string[]; counts: Counts } {

    // NOTE: update this for custom filters
    const allFiles = readAllFiles(srcDir, [(_fpath) => {
      // const fname = path.basename(fpath);
      // return minimatch(fpath, "tags.**.md")
      return true
    }]);
    logger.info(`Found ${allFiles.length} files in ${srcDir}`)
    const merged: string[] = [];

    for (const fileName of allFiles) {
      const srcFilePath = path.join(srcDir, fileName);
      const ext = path.extname(fileName);

      // we found a file that doesn't exist in the target, copy it over
      if (!fs.existsSync(path.join(destDir, fileName))) {
        const content = fs.readFileSync(srcFilePath, "utf-8");

        // if md, add frontmatter
        if (ext === ".md") {
          const { data: metadata, content: fileContent } = matter(content);
          Object.assign(metadata, extraMetadataOnTarget);
          if (!dryRun) writeWithFrontmatter(fileName, metadata, fileContent, destDir);
        } else {
          // otherwise, just copy
          if (!dryRun) fs.writeFileSync(path.join(destDir, fileName), content);
        }
      } else {
        // fileName exists, figure out merge
        const fSrcContents = fs.readFileSync(
          path.join(srcDir, fileName),
          "utf-8"
        );
        const fTargetContents = fs.readFileSync(
          path.join(destDir, fileName),
          "utf-8"
        );
        const mergedContents = fTargetContents + fSrcContents;
        if (!dryRun) {
          fs.writeFileSync(path.join(destDir, fileName), mergedContents);
        }
        merged.push(fileName);
      }
    }
    const counts: Counts = { allFiles: allFiles.length, merged: merged.length };
    return { merged, counts };
  }

}


function readAllFiles(
  targetDir: string,
  filterFns: ((filename: string) => boolean)[] = []
): string[] {
  const BLACKLIST = [".DS_Store", ".dendron.cache.json", "root.schema.yml", "package.json", ".gitignore"];
  const allFiles = fs.readdirSync(targetDir);
  const filteredFiles = allFiles.filter((filename) => {
    const filePath = path.join(targetDir, filename);
    const isFile = fs.statSync(filePath).isFile();
    const isInBlacklist = BLACKLIST.includes(filename);
    return filterFns.reduce(
      (prev, currFn) => prev && currFn(filePath),
      isFile && !isInBlacklist
    );
  });
  return filteredFiles;
}

function writeWithFrontmatter(
  filePath: string,
  metadata: any,
  content: string,
  to: string
): void {
  const matterResult = matter.stringify(content, metadata);
  fs.writeFileSync(path.join(to, filePath), matterResult);
}