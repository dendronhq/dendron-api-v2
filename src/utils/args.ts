import _ from "lodash";

export type ExcludeOption = {
  tags: { key: string; value: any }[];
  hierarchies: string[];
};

export const processExcludeOption = (value: string): ExcludeOption => {
  const out = value
    .split(",")
    .reduce((acc: Partial<ExcludeOption>, param: string) => {
      const [key, value] = param.split("=");
      let cvalue: any = value;
      switch (key) {
        case "tags": {
          const [tagKey, tagValue] = value.split(":");
          cvalue = [{ key: tagKey, tagValue }];
          acc[key] = cvalue;
          break;
        }
        case "hierarchies":
          cvalue = value.split("|");
          acc[key] = cvalue;
          break;
        default:
          throw new Error(`Invalid key "${key}"`);
      }
      return acc;
    }, {});
  return _.defaults(out, { tags: [], hierarchies: [] });
};
