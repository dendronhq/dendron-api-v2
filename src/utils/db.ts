import _ from "lodash";
import { Sequelize } from "sequelize";

export function sqlClient(storage: string, opts?: { logging?: boolean }) {
  const { logging } = _.defaults(opts, { logging: false });
  const sequelize = new Sequelize("database", "username", "password", {
    dialect: "sqlite",
    storage,
    logging,
  });
  return sequelize;
}
