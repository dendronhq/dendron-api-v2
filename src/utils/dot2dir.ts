import fs from "fs-extra";
import matter from "gray-matter";
import _ from "lodash";
import path from "path";
import { FileData } from "../types";

export type FileType = "file" | "index";

export function fnames2FileTypeMap(filenames: string[]): Map<string, FileType> {
  const fileTypes = new Map<string, FileType>();

  // Populate fileTypes with all filenames as keys and "file" as default value
  for (const filename of filenames) {
    fileTypes.set(filename, "file");
  }

  for (const filename of filenames) {
    const parts = filename.split(".");

    // Iterate through prefixes and update fileTypes if a parent is found
    for (let i = parts.length - 1; i > 0; i--) {
      const prefix = parts.slice(0, i).join(".");

      if (fileTypes.has(prefix)) {
        fileTypes.set(prefix, "index");
        break;
      }
    }
  }

  return fileTypes;
}

export function fname2FilePath(fname: string, fileType: FileType) {
  const _path = fname.replace(/\./g, '/');
  if (fileType === "file") {
    return _path + ".md"
  } else {
    return _path + "/index.md"
  }
}

/**
 *  Materialize files and folders in the file system
 */
export async function materializeFnames2FilesAndFolders(fileMappings: Map<string, FileType>, opts: { fnameDataMapping: Map<string, FileData>, baseDir: string }) {
  const fileMappingArray = Array.from(fileMappings.entries());
  await Promise.all(_.map(fileMappingArray, ([fname, fileType]) => {
    const _fpath = path.join(opts.baseDir, fname2FilePath(fname, fileType))
    return fs.ensureDir(path.dirname(_fpath))
  }));
  // // create all folders
  // await Promise.all(fileMappings.map(fpath => {
  //   return fs.ensureDir(path.dirname(fpath))
  // }));

  // create all files
  await Promise.all(_.map(fileMappingArray, ([fname, fileType]) => {
    const _fpath = path.join(opts.baseDir, fname2FilePath(fname, fileType))

    const { content, data } = opts.fnameDataMapping.get(fname)!
    const matterResult = matter.stringify(content, data);
    return fs.writeFile(_fpath, matterResult)
  }));
}

export function readFilesRecursively(dir: string,
  filterFns: ((filename: string) => boolean)[] = []): string[] {
  const files: string[] = [];

  function readDirRecursive(directory: string, currentPath = ''): void {
    const items = fs.readdirSync(directory);
    for (const item of items) {
      const itemPath = path.join(directory, item);
      const stat = fs.statSync(itemPath);
      if (stat.isDirectory()) {
        const nextPath = path.join(currentPath, item);
        readDirRecursive(itemPath, nextPath);
      } else if (stat.isFile()) {
        const filePath = path.join(currentPath, item);
        const shouldInclude = filterFns.reduce(
          (prev, currFn) => prev && currFn(filePath),
          true
        );
        if (shouldInclude) {
          files.push(filePath);
        }
      }
    }
  }

  readDirRecursive(dir);
  return files;
}

export function file2note(fpath: string) {
  const contents = fs.readFileSync(fpath, "utf8");
  const fname = path.basename(fpath).replace(/\.md$/, "")
  return { ...matter(contents), fname, fpath };
}