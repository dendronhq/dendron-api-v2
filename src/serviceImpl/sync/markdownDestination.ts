import { SyncToRequest, SyncToResponse } from "../../api/generated/api";
import { FileData, MatterFiles } from "../../types";
import { fnames2FileTypeMap, materializeFnames2FilesAndFolders } from "../../utils/dot2dir";

export class MarkdownDestination {
  async sync(files: MatterFiles[], opts: SyncToRequest): Promise<SyncToResponse> {
    const fnames = files.map(file => file.fname)
    const fnameFileTypeMapping = fnames2FileTypeMap(fnames);

    const baseDir = opts.dest;
    const fnameDataMapping = new Map<string, FileData>();
    files.forEach(({ fname, content, data }) => {
      fnameDataMapping.set(fname, { content, data })
    });

    await materializeFnames2FilesAndFolders(fnameFileTypeMapping, { baseDir, fnameDataMapping })
    return { numSynced: fnames.length }
  }
}