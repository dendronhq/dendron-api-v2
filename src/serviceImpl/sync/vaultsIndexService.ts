import { PrismaClient } from "@prisma/client";
import _ from "lodash";
import { minimatch } from "minimatch";
import { IndexVaultsRequest } from "../../api/generated/api";
import { logger } from "../../logger";
import { file2note, readFilesRecursively } from "../../utils/dot2dir";
import { NoteUtils } from "../../utils/note";

const parseTagsFromContent = (body: string): string[] => {
  const tags = body.match(/#[\w\.]+/g) || [];
  return tags.map((tag) => tag.slice(1));
}

//filterFns: ((filename: string) => boolean)[] = []): string[] {
const filterByHierarchies = (hierarchies: string[]) => {
  return (fname: string) => {
    return _.some(hierarchies, (hierarchyMatchPattern) => {
      return minimatch(fname, hierarchyMatchPattern);
    });
  }
}

// TODO: doesn't handle deleted records
export class VaultsIndexService {
  async execute(args: IndexVaultsRequest) {
    const ctx = "VaultsIndexService";
    logger.info({ ctx, msg: "enter", args });
    const dest = args.dest;
    const include = args.include || { hierarchies: [] };
    const pclient = new PrismaClient()
    const vaultName = args.vaultName;


    const notes = readFilesRecursively(args.src, [filterByHierarchies(include.hierarchies)])
      .filter((file) => file.endsWith(".md"))
      .map((file) => {
        const fpath = `${args.src}/${file}`;
        const note = file2note(fpath);
        return note;
      });

    logger.info({ ctx, msg: "fin:readFiles", numFiles: notes.length });


    if (args.purge) {
      logger.info({ ctx, msg: "purging..." })
      await pclient.note.deleteMany();
    }

    for (const note of notes) {
      const data = NoteUtils.getOrFillData(note);
      const { content, fname } = note;
      const { id, title, created, updated, tags } = data;

      const tagsFromContent = parseTagsFromContent(content);
      tagsFromContent.forEach((tag) => {
        if (!tags.includes(tag)) {
          tags.push(tag)
        }
      })

      const tableData = {
        id, title,
        created: Math.round(created / 1000),
        updated: Math.round(updated / 1000),
        tags: JSON.stringify(tags),
        fname,
        vault_name: vaultName,
        body: content
      }
      try {
        await pclient.note.upsert({ create: tableData, update: tableData, where: { id } })
      } catch (e) {
        logger.error({ ctx, msg: "error", e, note: { fname, data } })
        throw e;
      }
    }

    logger.info({ ctx, msg: "fin:readFiles", numFiles: notes.length });

  }
}