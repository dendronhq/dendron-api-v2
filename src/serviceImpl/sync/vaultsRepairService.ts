import execa from "execa";
import _ from "lodash";
import path from "path";
import { RepairVaultsRequest } from "../../api/generated/api";
import { logger } from "../../logger";
import { MatterFile } from "../../types";
import { file2note, readFilesRecursively } from "../../utils/dot2dir";
import { NoteUtils } from "../../utils/note";

// TODO: properly get updated/created
function getCreatedFromGit(fpath: string) {
  const fbasename = path.basename(fpath)
  const resp = execa.commandSync(`git log --diff-filter=A --follow --format=%aD -1 -- ${fbasename} `, {
    cwd: path.dirname(fpath)
  }).stdout
  return Date.parse(resp)
}

function getUpdatedFromGit(fpath: string) {
  const fbasename = path.basename(fpath)
  const resp = execa.commandSync(`git log --format=%aD -1 -- ${fbasename} `, {
    cwd: path.dirname(fpath)
  }).stdout
  return Date.parse(resp)
}


/**
 * See pkg.dendronv2-api.vaults.repair.md
 */
export class VaultsRepairService {
  async execute(args: RepairVaultsRequest) {
    const ctx = "VaultsIndexService";
    logger.info({ ctx, msg: "enter", args });

    const needsRepair = [] as MatterFile[];
    const allFiles = readFilesRecursively(args.src)
      .filter((file) => file.endsWith(".md"));

    let progress = 0;
    const totalFiles = allFiles.length;
    const increment = totalFiles / 10;

    function repairNotes() {
      for (const file of allFiles) {
        const fpath = `${args.src}/${file}`;
        const note = file2note(fpath);
        let repaired = false;

        if (note.data.id === undefined) {
          repaired = true;
          note.data.id = NoteUtils.genId();
        }

        if (note.data.title === undefined || note.data.title === '__MISSING__' || note.data.title === null || !_.isString(note.data.title)) {
          repaired = true;
          note.data.title = NoteUtils.genTitle(note);
        }

        if (note.data.updated === undefined) {
          repaired = true;
          note.data.updated = 1;
        }

        if (note.data.created === undefined) {
          repaired = true;
          // TODO: figure this out later
          note.data.created = 1;
        }

        if (!_.isArray(note.data.tags)) {
          repaired = true;
          // TODO: figure this out later
          note.data.tags = []
        }

        if (repaired) {
          needsRepair.push(note);
        }

        progress += 1;
        if (progress % increment === 0) {
          logger.info({
            ctx,
            msg: `Repair progress: ${progress * 100 / totalFiles}%`,
          });
        }
      }
    }

    // Call the async function
    repairNotes();


    logger.info({ ctx, msg: "repairing", numNotes: needsRepair.length });
    await Promise.all(_.map(needsRepair, async (note) => {
      return NoteUtils.write(note, { fpath: note.fpath })
    }));


  }
}