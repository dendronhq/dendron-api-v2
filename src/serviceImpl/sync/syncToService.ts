import fs from "fs-extra";
import matter from "gray-matter";
import _ from "lodash";
import { minimatch } from "minimatch";
import * as z from "zod";
import { SyncToRequest } from "../../api/generated/api";
import { logger } from "../../logger";
import { MarkdownDestination } from "./markdownDestination";

type IncludeOption = {
  hierarchies: string[];
};

type ExcludeOption = {
  tags: { key: string; value: any }[];
  hierarchies: string[];
};

const processIncludeOption = (value: string): IncludeOption => {
  const out = value
    .split(",")
    .reduce((acc: Partial<IncludeOption>, param: string) => {
      const [key, value] = param.split("=");
      let cvalue: any = value;
      switch (key) {
        case "hierarchies":
          cvalue = value.split("|");
          acc[key] = cvalue;
          break;
        default:
          throw new Error(`Invalid key "${key}"`);
      }
      return acc;
    }, {});
  return _.defaults(out, { hierarchies: ["*"] });
};

const processExcludeOption = (value: string): ExcludeOption => {
  const out = value
    .split(",")
    .reduce((acc: Partial<ExcludeOption>, param: string) => {
      const [key, value] = param.split("=");
      let cvalue: any = value;
      switch (key) {
        case "tags": {
          const [tagKey, tagValue] = value.split(":");
          cvalue = [{ key: tagKey, tagValue }];
          acc[key] = cvalue;
          break;
        }
        case "hierarchies":
          cvalue = value.split("|");
          acc[key] = cvalue;
          break;
        default:
          throw new Error(`Invalid key "${key}"`);
      }
      return acc;
    }, {});
  return _.defaults(out, { tags: [], hierarchies: [] });
};

const optionsSchema = z.object({
  include: z.string().transform(processIncludeOption),
  exclude: z.string().transform(processExcludeOption),
  targetFormat: z.string(),
  src: z.string(),
  dest: z.string(),
  deleteMissing: z.boolean().default(false),
});

export class SyncToService {
  async execute(args: SyncToRequest) {
    const ctx = "SyncToService";
    const _args = optionsSchema.parse(args);
    logger.info({ ctx, msg: "enter", args: _args });

    // Read all files with frontmatter using the "gray-matter" library
    const files = fs
      .readdirSync(args.src)
      .filter((file) => file.endsWith(".md"))
      .map((file) => {
        const contents = fs.readFileSync(`${args.src}/${file}`, "utf8");
        const fname = file.replace(/\.md$/, "");
        return { ...matter(contents), fname, fpath: `${args.src}/${file}` };
      });
    logger.info({ ctx, msg: "fin:readFiles", numFiles: files.length });

    // const matchOnHierarchy = (opts: {hierarchies: string[], files: (typeof files)}) => {
    //   let filesToSync = opts.hierarchies.flatMap(hierarchyMatchPattern => {
    //     return opts.files.filter(file => minimatch(file.fname, hierarchyMatchPattern));
    //   }) as unknown as (typeof files)
    // }

    // include
    const hierarchies = _args.include.hierarchies;
    let filesToSync = hierarchies.flatMap((hierarchyMatchPattern) => {
      return files.filter((file) =>
        minimatch(file.fname, hierarchyMatchPattern)
      );
    }) as unknown as typeof files;
    logger.info({
      ctx,
      msg: "fin:filterHierarchies",
      numFiles: filesToSync.length,
    });

    // exclude
    filesToSync = _args.exclude.hierarchies.flatMap((hierarchyMatchPattern) => {
      return filesToSync.filter(
        (file) => !minimatch(file.fname, hierarchyMatchPattern)
      );
    }) as unknown as typeof files;
    // _args.exclude.tags.forEach(tag => {
    //   filesToSync = filesToSync.filter(file => file.data[tag.key] !== tag.value);
    // });
    // awlays exclude `public: false`
    // filesToSync = filesToSync.filter(file => (!('public' in file.data) || file.data['public']));
    logger.info({ ctx, msg: "fin:excludeTags", numFiles: filesToSync.length });

    logger.info({
      ctx,
      matches: filesToSync.map((file) => file.fname).join("\n"),
    });
    // TODO: support incremental sync

    if (_args.deleteMissing) {
      logger.info({ ctx, msg: "deleteMissing", dest: _args.dest });
      fs.emptyDirSync(_args.dest);
    }

    switch (args.targetFormat) {
      case "markdown": {
        const dest = new MarkdownDestination();
        return dest.sync(filesToSync, args);
      }
      default:
        throw new Error(
          `Error: "${args.targetFormat}" is not a supported exclude parameter.`
        );
    }
  }
}
