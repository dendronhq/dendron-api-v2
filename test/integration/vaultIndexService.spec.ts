import { PrismaClient } from '@prisma/client';
import { VaultsIndexService } from '../../src/serviceImpl/sync/vaultsIndexService';
import { file2note, readFilesRecursively } from "../../src/utils/dot2dir";

jest.mock("../../src/utils/dot2dir");

let pclient: PrismaClient;

beforeAll(async () => {
  pclient = new PrismaClient();
  await pclient.note.deleteMany();
});

afterAll(async () => {
  await pclient.$disconnect();
});

describe('when calling VaultsIndexService', () => {
  describe("when parsing tags", () => {
    it('should parse tags', async () => {
      // Mock the utilities
      (readFilesRecursively as jest.Mock).mockReturnValue(['test.md']);
      (file2note as jest.Mock).mockReturnValue({
        content: 'test content #foo',
        fname: 'test.md',
        data: {
          id: '1',
          title: 'Test Note',
          created: Date.now(),
          updated: Date.now(),
          tags: ['test'],
        },
      });

      const service = new VaultsIndexService();
      const result = await service.execute({
        vaultName: 'testVault',
        src: '/path/to/src',
        include: { hierarchies: ['*'] },
        dest: ""
      });

      // Check the results
      expect(result).toEqual({ numNotesIndexed: 1 });

      // Check that the note has been stored in the database
      const dbNote = await pclient.note.findUnique({ where: { id: '1' } });
      expect(dbNote).not.toBeNull();
      expect(dbNote?.title).toBe('Test Note');
      expect(dbNote?.tags).toEqual(JSON.stringify(["test", "foo"]))
      // expect(dbNote).toMatchSnapshot()
    });
  });

  describe("when parsing tags with number", () => {
    it('should parse tags', async () => {
      // Mock the utilities
      (readFilesRecursively as jest.Mock).mockReturnValue(['test.md']);
      (file2note as jest.Mock).mockReturnValue({
        content: 'test content #foo.1',
        fname: 'test.md',
        data: {
          id: '1',
          title: 'Test Note',
          created: Date.now(),
          updated: Date.now(),
          tags: ['test'],
        },
      });

      const service = new VaultsIndexService();
      const result = await service.execute({
        vaultName: 'testVault',
        src: '/path/to/src',
        include: { hierarchies: ['*'] },
        dest: ""
      });

      // Check the results
      expect(result).toEqual({ numNotesIndexed: 1 });

      // Check that the note has been stored in the database
      const dbNote = await pclient.note.findUnique({ where: { id: '1' } });
      expect(dbNote).not.toBeNull();
      expect(dbNote?.title).toBe('Test Note');
      expect(dbNote?.tags).toEqual(JSON.stringify(["test", "foo.1"]))
      expect(dbNote).toMatchSnapshot()
    });
  });

});
