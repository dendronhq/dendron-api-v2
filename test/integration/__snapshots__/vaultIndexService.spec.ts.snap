// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration Tests should return a successful response 1`] = `200`;

exports[`when calling VaultsIndexService when parsing tags should parse tags 1`] = `
{
  "body": "test content #foo",
  "created": 1686424899,
  "fname": "test.md",
  "id": "1",
  "tags": "["test","foo"]",
  "title": "Test Note",
  "updated": 1686424899,
  "vault_name": "testVault",
}
`;

exports[`when calling VaultsIndexService when parsing tags with number and dash should parse tags 1`] = `
{
  "body": "test content #foo.two-1",
  "created": 1686425313,
  "fname": "test.md",
  "id": "1",
  "tags": "["test","foo.two-1"]",
  "title": "Test Note",
  "updated": 1686425313,
  "vault_name": "testVault",
}
`;

exports[`when calling VaultsIndexService when parsing tags with number should parse tags 1`] = `
{
  "body": "test content #foo.1",
  "created": 1686425096,
  "fname": "test.md",
  "id": "1",
  "tags": "["test","foo.1"]",
  "title": "Test Note",
  "updated": 1686425096,
  "vault_name": "testVault",
}
`;
